src/ext/php7/engine_hooks.c

```diff
@@ -84,13 +84,39 @@ static bool _dd_should_trace_helper(zend_execute_data *call, zend_function *fbc,
      *
      * It would avoid lowering the string and reduce memory churn; win-win.
      */
-    *dispatch = ddtrace_find_dispatch(this ? Z_OBJCE_P(this) : fbc->common.scope, &fname);
-    return *dispatch;
+    zend_class_entry *scope = this ? Z_OBJCE_P(this) : fbc->common.scope;
+
+    ddtrace_dispatch_t *dispatch = ddtrace_find_dispatch(scope, &fname);
+    if (dispatch != NULL && dispatch->options & DDTRACE_DISPATCH_DEFERRED_LOADER) {
+        // don't execute in the future
+        dispatch->options ^= DDTRACE_DISPATCH_DEFERRED_LOADER;
```

Comment:
I think you mention that @labbati pointed out something along these lines, but would it make sense to remove `DDTRACE_DISPATCH_DEFERRED_LOADER`'s from the dispatch hash table after the first call and before loading the integration? I think there are a lot of places (most cases?) were we don't instrument the "signature"/loading call.

Reply:
In the initial design, the deferred loading was meant to only have 1:1 relation with functions which trigger loading to functions which are traced  - meaning all trigger function/methods were meant to be overwritten on the first load.

When that relation is broken, either intentionally or by mistake, we need to make sure the loading function is called only once. 

I'm currently thinking either about implementing a already_loaded hashmap - that would track all function callables which already were executed. And we would check that hashmap to see if we've already called the function.

Second idea is to be able to unload all functions which were loaded on first load. Currently it would probably require storing a function pointer in `dispatch struct`

Reply:
I'm leaning more towards "already_loaded" hashmap

Reply:
The deduplication will happen in a separate PR, right now there will need to be 1:1 relation to instrumented function and deferred functions.